% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={BCB 546 R assignment},
  pdfauthor={Prita},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{BCB 546 R assignment}
\author{Prita}
\date{3/18/2021}

\begin{document}
\maketitle

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# clear global environment}
\FunctionTok{rm}\NormalTok{(}\AttributeTok{list =} \FunctionTok{ls}\NormalTok{())}

\CommentTok{\# Load require packages}
\FunctionTok{library}\NormalTok{(tidyverse)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## -- Attaching packages --------------------------------------- tidyverse 1.3.0 --
\end{verbatim}

\begin{verbatim}
## v ggplot2 3.3.3     v purrr   0.3.4
## v tibble  3.1.0     v dplyr   1.0.5
## v tidyr   1.1.3     v stringr 1.4.0
## v readr   1.4.0     v forcats 0.5.1
\end{verbatim}

\begin{verbatim}
## -- Conflicts ------------------------------------------ tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(dplyr)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# download data files}
\NormalTok{fang }\OtherTok{\textless{}{-}} \FunctionTok{read\_tsv}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/EEOB{-}BioData/BCB546{-}Spring2021/main/assignments/UNIX\_Assignment/fang\_et\_al\_genotypes.txt"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## -- Column specification --------------------------------------------------------
## cols(
##   .default = col_character()
## )
## i Use `spec()` for the full column specifications.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(fang)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 986
##   Sample_ID JG_OTU   Group abph1.20 abph1.22 ae1.3 ae1.4 ae1.5 an1.4 ba1.6 ba1.9
##   <chr>     <chr>    <chr> <chr>    <chr>    <chr> <chr> <chr> <chr> <chr> <chr>
## 1 SL-15     T-aust-1 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   ?/?   G/G  
## 2 SL-16     T-aust-2 TRIPS ?/?      ?/?      T/T   ?/?   T/T   C/C   A/G   G/G  
## 3 SL-11     T-brav-1 TRIPS ?/?      ?/?      T/T   G/G   T/T   ?/?   G/G   G/G  
## 4 SL-12     T-brav-2 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   G/G   G/G  
## 5 SL-18     T-cund   TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   ?/?   G/G  
## 6 SL-2      T-dact-1 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   A/G   G/G  
## # ... with 975 more variables: bt2.5 <chr>, bt2.7 <chr>, bt2.8 <chr>,
## #   Fea2.1 <chr>, Fea2.5 <chr>, id1.3 <chr>, lg2.11 <chr>, lg2.2 <chr>,
## #   pbf1.1 <chr>, pbf1.2 <chr>, pbf1.3 <chr>, pbf1.5 <chr>, pbf1.6 <chr>,
## #   pbf1.7 <chr>, pbf1.8 <chr>, PZA00003.11 <chr>, PZA00004.2 <chr>,
## #   PZA00005.8 <chr>, PZA00005.9 <chr>, PZA00006.13 <chr>, PZA00006.14 <chr>,
## #   PZA00008.1 <chr>, PZA00010.5 <chr>, PZA00013.10 <chr>, PZA00013.11 <chr>,
## #   PZA00013.9 <chr>, PZA00015.4 <chr>, PZA00017.1 <chr>, PZA00018.5 <chr>,
## #   PZA00029.11 <chr>, PZA00029.12 <chr>, PZA00030.11 <chr>, PZA00031.5 <chr>,
## #   PZA00041.3 <chr>, PZA00042.2 <chr>, PZA00042.5 <chr>, PZA00043.7 <chr>,
## #   PZA00045.1 <chr>, PZA00047.2 <chr>, PZA00049.12 <chr>, PZA00050.9 <chr>,
## #   PZA00051.2 <chr>, PZA00058.5 <chr>, PZA00058.6 <chr>, PZA00060.2 <chr>,
## #   PZA00061.1 <chr>, PZA00065.2 <chr>, PZA00069.4 <chr>, PZA00070.5 <chr>,
## #   PZA00078.2 <chr>, PZA00079.1 <chr>, PZA00081.17 <chr>, PZA00084.2 <chr>,
## #   PZA00084.3 <chr>, PZA00086.8 <chr>, PZA00088.3 <chr>, PZA00090.2 <chr>,
## #   PZA00092.1 <chr>, PZA00092.5 <chr>, PZA00093.2 <chr>, PZA00096.26 <chr>,
## #   PZA00097.13 <chr>, PZA00098.14 <chr>, PZA00100.10 <chr>, PZA00100.12 <chr>,
## #   PZA00100.14 <chr>, PZA00100.9 <chr>, PZA00103.20 <chr>, PZA00106.9 <chr>,
## #   PZA00107.18 <chr>, PZA00108.12 <chr>, PZA00108.14 <chr>, PZA00108.15 <chr>,
## #   PZA00109.3 <chr>, PZA00109.5 <chr>, PZA00111.2 <chr>, PZA00111.4 <chr>,
## #   PZA00111.5 <chr>, PZA00111.6 <chr>, PZA00111.8 <chr>, PZA00114.3 <chr>,
## #   PZA00116.2 <chr>, PZA00119.4 <chr>, PZA00120.4 <chr>, PZA00123.1 <chr>,
## #   PZA00125.2 <chr>, PZA00131.14 <chr>, PZA00132.17 <chr>, PZA00132.18 <chr>,
## #   PZA00132.3 <chr>, PZA00135.6 <chr>, PZA00137.2 <chr>, PZA00139.14 <chr>,
## #   PZA00140.10 <chr>, PZA00140.6 <chr>, PZA00140.9 <chr>, PZA00142.6 <chr>,
## #   PZA00148.2 <chr>, PZA00153.3 <chr>, PZA00153.6 <chr>, ...
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{SNP }\OtherTok{\textless{}{-}} \FunctionTok{read\_tsv}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/EEOB{-}BioData/BCB546{-}Spring2021/main/assignments/UNIX\_Assignment/snp\_position.txt"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## -- Column specification --------------------------------------------------------
## cols(
##   SNP_ID = col_character(),
##   cdv_marker_id = col_double(),
##   Chromosome = col_character(),
##   Position = col_character(),
##   alt_pos = col_character(),
##   mult_positions = col_character(),
##   amplicon = col_character(),
##   cdv_map_feature.name = col_character(),
##   gene = col_character(),
##   `candidate/random` = col_character(),
##   Genaissance_daa_id = col_double(),
##   Sequenom_daa_id = col_double(),
##   count_amplicons = col_double(),
##   count_cmf = col_double(),
##   count_gene = col_double()
## )
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(SNP)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 6 x 15
##   SNP_ID   cdv_marker_id Chromosome Position  alt_pos mult_positions amplicon
##   <chr>            <dbl> <chr>      <chr>     <chr>   <chr>          <chr>   
## 1 abph1.20          5976 2          27403404  <NA>    <NA>           abph1   
## 2 abph1.22          5978 2          27403892  <NA>    <NA>           abph1   
## 3 ae1.3             6605 5          167889790 <NA>    <NA>           ae1     
## 4 ae1.4             6606 5          167889682 <NA>    <NA>           ae1     
## 5 ae1.5             6607 5          167889821 <NA>    <NA>           ae1     
## 6 an1.4             5982 1          240498509 <NA>    <NA>           an1     
## # ... with 8 more variables: cdv_map_feature.name <chr>, gene <chr>,
## #   candidate/random <chr>, Genaissance_daa_id <dbl>, Sequenom_daa_id <dbl>,
## #   count_amplicons <dbl>, count_cmf <dbl>, count_gene <dbl>
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data inspection}

\CommentTok{\# view(fang)}
\CommentTok{\# vieq(SNP)}

\CommentTok{\#str(fang)}
\CommentTok{\#str(SNP)}

\FunctionTok{dim}\NormalTok{(fang)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 2782  986
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(SNP)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 983  15
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{class}\NormalTok{(fang)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "spec_tbl_df" "tbl_df"      "tbl"         "data.frame"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{class}\NormalTok{(SNP)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "spec_tbl_df" "tbl_df"      "tbl"         "data.frame"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{fang[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 2,782 x 10
##    Sample_ID JG_OTU   Group abph1.20 abph1.22 ae1.3 ae1.4 ae1.5 an1.4 ba1.6
##    <chr>     <chr>    <chr> <chr>    <chr>    <chr> <chr> <chr> <chr> <chr>
##  1 SL-15     T-aust-1 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   ?/?  
##  2 SL-16     T-aust-2 TRIPS ?/?      ?/?      T/T   ?/?   T/T   C/C   A/G  
##  3 SL-11     T-brav-1 TRIPS ?/?      ?/?      T/T   G/G   T/T   ?/?   G/G  
##  4 SL-12     T-brav-2 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   G/G  
##  5 SL-18     T-cund   TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   ?/?  
##  6 SL-2      T-dact-1 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   A/G  
##  7 SL-4      T-dact-2 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   G/G  
##  8 SL-6      T-dact-3 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   ?/?  
##  9 SL-7      T-dact-4 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   G/G  
## 10 SL-5      T-dact-5 TRIPS ?/?      ?/?      T/T   G/G   T/T   C/C   ?/?  
## # ... with 2,772 more rows
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{SNP[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 983 x 10
##    SNP_ID   cdv_marker_id Chromosome Position  alt_pos mult_positions amplicon
##    <chr>            <dbl> <chr>      <chr>     <chr>   <chr>          <chr>   
##  1 abph1.20          5976 2          27403404  <NA>    <NA>           abph1   
##  2 abph1.22          5978 2          27403892  <NA>    <NA>           abph1   
##  3 ae1.3             6605 5          167889790 <NA>    <NA>           ae1     
##  4 ae1.4             6606 5          167889682 <NA>    <NA>           ae1     
##  5 ae1.5             6607 5          167889821 <NA>    <NA>           ae1     
##  6 an1.4             5982 1          240498509 <NA>    <NA>           an1     
##  7 ba1.6             3463 3          181362952 <NA>    <NA>           ba1     
##  8 ba1.9             3466 3          181362666 <NA>    <NA>           ba1     
##  9 bt2.5             5983 4          66290049  <NA>    <NA>           bt2     
## 10 bt2.7             5985 4          66290994  <NA>    <NA>           bt2     
## # ... with 973 more rows, and 3 more variables: cdv_map_feature.name <chr>,
## #   gene <chr>, candidate/random <chr>
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data processing I}


\CommentTok{\# transpose the fang et al file as data frame }
\NormalTok{tfang }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{t}\NormalTok{(fang))}

\CommentTok{\# Make smaple names to column names}
\FunctionTok{names}\NormalTok{(tfang) }\OtherTok{\textless{}{-}} \FunctionTok{lapply}\NormalTok{(tfang[}\DecValTok{1}\NormalTok{, ], as.character)}

\CommentTok{\# Remove sample names in rows}
\NormalTok{tfang }\OtherTok{\textless{}{-}}\NormalTok{ tfang[}\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{,]}

\CommentTok{\# To make column of the row names}
\NormalTok{tfang }\OtherTok{\textless{}{-}} \FunctionTok{rownames\_to\_column}\NormalTok{(tfang, }\AttributeTok{var=}\StringTok{"SNP\_ID"}\NormalTok{)}

\CommentTok{\# create SNP file with SNP\_ID, Chromosome and position}
\NormalTok{SNP\_edit }\OtherTok{\textless{}{-}}\NormalTok{ SNP }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{select}\NormalTok{(SNP\_ID, Chromosome, Position)}

\CommentTok{\# Merge both files by SNP\_ID}
\NormalTok{merged\_fangSNP }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(SNP\_edit,tfang, }\AttributeTok{by.x=}\StringTok{"SNP\_ID"}\NormalTok{, }\AttributeTok{by.y=}\StringTok{"SNP\_ID"}\NormalTok{, }\AttributeTok{all =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{view}\NormalTok{(merged\_fangSNP)}

\CommentTok{\# select groups from the merged file}

\DocumentationTok{\#\# Maize {-} ZMMIL, ZMMLR, ZMMMR}

\NormalTok{Maize }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(fang, Group }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"ZMMLR"}\NormalTok{, }\StringTok{"ZMMMR"}\NormalTok{, }\StringTok{"ZMMIL"}\NormalTok{))}
\NormalTok{Maize\_name }\OtherTok{\textless{}{-}}\NormalTok{ Maize[,}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{)]}
\NormalTok{Maize\_fullname }\OtherTok{\textless{}{-}} \FunctionTok{apply}\NormalTok{(Maize\_name, }\DecValTok{1}\NormalTok{, paste, }\AttributeTok{collapse =} \StringTok{", "}\NormalTok{) }

\DocumentationTok{\#\# Teosinate {-} ZMPBA, ZMPIL, ZMPJA}

\NormalTok{Teosinate }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(fang, Group }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"ZMPBA"}\NormalTok{, }\StringTok{"ZMPIL"}\NormalTok{,}\StringTok{"ZMPJA"}\NormalTok{))}
\NormalTok{Teosinate\_name }\OtherTok{\textless{}{-}}\NormalTok{ Teosinate[,}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{)]}
\NormalTok{Teosinate\_fullname }\OtherTok{\textless{}{-}} \FunctionTok{apply}\NormalTok{(Teosinate\_name, }\DecValTok{1}\NormalTok{, paste, }\AttributeTok{collapse =}\StringTok{","}\NormalTok{)}

\CommentTok{\# edit maize files {-} transpose and merge}

\NormalTok{tMaize}\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{t}\NormalTok{(Maize[,}\FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\DecValTok{1}\SpecialCharTok{:{-}}\DecValTok{3}\NormalTok{)]))}
\FunctionTok{colnames}\NormalTok{(tMaize) }\OtherTok{\textless{}{-}}\NormalTok{ Maize\_fullname}
\NormalTok{rnames }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(tMaize))}
\FunctionTok{rownames}\NormalTok{(tMaize) }\OtherTok{\textless{}{-}} \ConstantTok{NULL}
\NormalTok{tMaize }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(rnames,tMaize)}
\FunctionTok{colnames}\NormalTok{(tMaize)[}\DecValTok{1}\NormalTok{] }\OtherTok{\textless{}{-}} \StringTok{"SNP\_ID"}

\FunctionTok{view}\NormalTok{(tMaize)}

\NormalTok{merge\_Maize }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(SNP\_edit,tMaize, }\AttributeTok{by =} \StringTok{"SNP\_ID"}\NormalTok{)}
\FunctionTok{view}\NormalTok{(merge\_Maize)}

\CommentTok{\# edit Teosinate files {-} transpose and merge}

\NormalTok{tTeosinate}\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{t}\NormalTok{(Teosinate[,}\FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\DecValTok{1}\SpecialCharTok{:{-}}\DecValTok{3}\NormalTok{)]))}
\FunctionTok{colnames}\NormalTok{(tTeosinate) }\OtherTok{\textless{}{-}}\NormalTok{ Teosinate\_fullname}
\NormalTok{rnames }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(tTeosinate))}
\FunctionTok{rownames}\NormalTok{(tTeosinate) }\OtherTok{\textless{}{-}} \ConstantTok{NULL}
\NormalTok{tTeosinate }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(rnames,tTeosinate)}
\FunctionTok{colnames}\NormalTok{(tTeosinate)[}\DecValTok{1}\NormalTok{] }\OtherTok{\textless{}{-}} \StringTok{"SNP\_ID"}

\FunctionTok{view}\NormalTok{(tTeosinate)}

\NormalTok{merge\_Teosinate }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(SNP\_edit, tTeosinate, }\AttributeTok{by =} \StringTok{"SNP\_ID"}\NormalTok{)}
\FunctionTok{view}\NormalTok{(merge\_Teosinate)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# Data processing I}
\CommentTok{\# Editing files }

\FunctionTok{library}\NormalTok{(tidyverse)}
\FunctionTok{library}\NormalTok{(dplyr)}

\CommentTok{\# create directories}
\FunctionTok{dir.create}\NormalTok{(}\StringTok{"./Maize\_files"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in dir.create("./Maize_files"): '.\Maize_files' already exists
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dir.create}\NormalTok{(}\StringTok{"./Teosinate\_files"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in dir.create("./Teosinate_files"): '.\Teosinate_files' already exists
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dir.create}\NormalTok{(}\StringTok{"./Graphs\_files"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in dir.create("./Graphs_files"): '.\Graphs_files' already exists
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# replace the missing values by ? and {-} for decreasing order files}

\NormalTok{Maize\_replaced }\OtherTok{\textless{}{-}}\NormalTok{ merge\_Maize }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate\_all}\NormalTok{(}\ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{?"}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{{-}"}\NormalTok{,x))}
\NormalTok{Teosinate\_replaced }\OtherTok{\textless{}{-}}\NormalTok{ merge\_Teosinate }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate\_all}\NormalTok{(}\ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{?"}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{{-}"}\NormalTok{,x))}

\DocumentationTok{\#\# sorting in ascending }

\NormalTok{merge\_Maize }\OtherTok{\textless{}{-}}\NormalTok{ merge\_Maize }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{arrange}\NormalTok{(.,}\FunctionTok{as.numeric}\NormalTok{(Position))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{merge\_Teosinate }\OtherTok{\textless{}{-}}\NormalTok{ merge\_Teosinate }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{arrange}\NormalTok{(.,}\FunctionTok{as.numeric}\NormalTok{(Position))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#merge\_Maize \textless{}{-} merge\_Maize[order(merge\_Maize$Position)]}
\CommentTok{\#merge\_Teosinate \textless{}{-} merge\_Teosinate[order(merge\_Teosinate$Position)]}


\CommentTok{\# sorting in descending (use \_replaced files to sort in descending order)}
\NormalTok{Maize\_replaced }\OtherTok{\textless{}{-}}\NormalTok{ Maize\_replaced }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{arrange}\NormalTok{(.,}\FunctionTok{desc}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(Position)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Teosinate\_replaced }\OtherTok{\textless{}{-}}\NormalTok{ Teosinate\_replaced }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{arrange}\NormalTok{(.,}\FunctionTok{desc}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(Position)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Maize\_replaced \textless{}{-} Maize\_replaced[order({-}merge$Position)]}
\CommentTok{\# Teosinate\_replaced \textless{}{-} Teosinate\_replaced[order({-}merge$Position)]}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Using for loop to create each {-} 10 chromosome files}
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{) \{}
\NormalTok{  Maize\_Asc }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(merge\_Maize, Chromosome }\SpecialCharTok{==}\NormalTok{ i)}
\NormalTok{  Maize\_Desc }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(Maize\_replaced, Chromosome }\SpecialCharTok{==}\NormalTok{ i)}
  
  \FunctionTok{write\_tsv}\NormalTok{(Maize\_Asc, }\FunctionTok{file.path}\NormalTok{(}\StringTok{"./Maize\_files"}\NormalTok{, }\FunctionTok{paste}\NormalTok{(}\StringTok{"Maize\_Chr"}\NormalTok{,i,}\StringTok{"increasing.txt"}\NormalTok{, }\AttributeTok{sep =} \StringTok{"\_"}\NormalTok{)))}
  \FunctionTok{write\_tsv}\NormalTok{(Maize\_Desc, }\FunctionTok{file.path}\NormalTok{(}\StringTok{"./Maize\_files"}\NormalTok{, }\FunctionTok{paste}\NormalTok{(}\StringTok{"Maize\_Chr"}\NormalTok{,i, }\StringTok{"decreasing.txt"}\NormalTok{, }\AttributeTok{sep =} \StringTok{"\_"}\NormalTok{)))}
  
\NormalTok{  Teosinate\_Asc }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(merge\_Teosinate, Chromosome }\SpecialCharTok{==}\NormalTok{ i)}
\NormalTok{  Teosinate\_Desc }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(Teosinate\_replaced, Chromosome }\SpecialCharTok{==}\NormalTok{ i)}
  
  \FunctionTok{write\_tsv}\NormalTok{(Teosinate\_Asc, }\FunctionTok{file.path}\NormalTok{(}\StringTok{"./Teosinate\_files"}\NormalTok{, }\FunctionTok{paste}\NormalTok{(}\StringTok{"Teosinate\_Chr"}\NormalTok{,i,}\StringTok{"increasing.txt"}\NormalTok{ , }\AttributeTok{sep =} \StringTok{"\_"}\NormalTok{)))}
  \FunctionTok{write\_tsv}\NormalTok{(Teosinate\_Desc, }\FunctionTok{file.path}\NormalTok{(}\StringTok{"./Teosinate\_files"}\NormalTok{ , }\FunctionTok{paste}\NormalTok{(}\StringTok{"Teosinate\_Chr"}\NormalTok{,i, }\StringTok{"decreasing.txt"}\NormalTok{, }\AttributeTok{sep =} \StringTok{"\_"}\NormalTok{)))}
  
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# Data Visulization}

\CommentTok{\# Total SNPs per chromosome}

\NormalTok{filtered\_fangSNP }\OtherTok{\textless{}{-}} \FunctionTok{filter}\NormalTok{(merged\_fangSNP , Position }\SpecialCharTok{!=} \StringTok{"multiple"}\NormalTok{,  Position }\SpecialCharTok{!=} \StringTok{"unknown"}\NormalTok{)}

\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data =}\NormalTok{ filtered\_fangSNP) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{mapping =} \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ Chromosome)) }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"SNPs per chromosome"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{xlab}\NormalTok{(}\AttributeTok{label =} \StringTok{"Chromosome"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ylab}\NormalTok{(}\AttributeTok{label =} \StringTok{"Number of SNPs"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_bw}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\includegraphics{BCB-546-R_files/figure-latex/unnamed-chunk-7-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\AttributeTok{filename =} \StringTok{"./Graphs\_files/SNPs per chromosome.png"}\NormalTok{, }\AttributeTok{device =} \StringTok{"png"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Saving 6.5 x 4.5 in image
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# Generate density plot to show distribution}

\FunctionTok{ggplot}\NormalTok{(filtered\_fangSNP, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=} \FunctionTok{as.numeric}\NormalTok{(Position))) }\SpecialCharTok{+} \FunctionTok{geom\_density}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{fill =}\NormalTok{ Chromosome)) }\SpecialCharTok{+} \FunctionTok{facet\_wrap}\NormalTok{(}\SpecialCharTok{\textasciitilde{}}\NormalTok{ Chromosome) }\SpecialCharTok{+} \FunctionTok{theme\_linedraw}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Position"}\NormalTok{, }\AttributeTok{y =} \StringTok{"Density"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{BCB-546-R_files/figure-latex/unnamed-chunk-7-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\AttributeTok{filename =} \StringTok{"./Graphs\_files/SNP Distribution.png"}\NormalTok{ , }\AttributeTok{device =} \StringTok{"png"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Saving 6.5 x 4.5 in image
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# Data processing II for further grouping}
\NormalTok{tidy\_fang }\OtherTok{\textless{}{-}}\NormalTok{ fang }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{select}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{JG\_OTU) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{pivot\_longer}\NormalTok{( }\SpecialCharTok{{-}}\NormalTok{Sample\_ID}\SpecialCharTok{:{-}}\NormalTok{Group, }\AttributeTok{names\_to =} \StringTok{"SNP\_ID"}\NormalTok{, }\AttributeTok{values\_to =} \StringTok{"Sequence"}\NormalTok{)}

\NormalTok{tidy\_fang }\OtherTok{\textless{}{-}}\NormalTok{ tidy\_fang }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{new\_sequence =} \FunctionTok{ifelse}\NormalTok{(Sequence }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"A/A"}\NormalTok{,}\StringTok{"T/T"}\NormalTok{,}\StringTok{"C/C"}\NormalTok{,}\StringTok{"G/G"}\NormalTok{), }\StringTok{"Homozygous"}\NormalTok{, }
  \FunctionTok{ifelse}\NormalTok{(Sequence }\SpecialCharTok{==} \StringTok{"?/?"}\NormalTok{, }\StringTok{"Missing"}\NormalTok{,}\StringTok{"Heterozygous"}\NormalTok{)))}

\DocumentationTok{\#\# SNPs/ Zygosity in sample}


\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data =}\NormalTok{ tidy\_fang, }\AttributeTok{mapping =} \FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{ Sample\_ID, }\AttributeTok{fill =}\NormalTok{ new\_sequence)) }\SpecialCharTok{+} \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{position =} \StringTok{"stack"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{theme\_classic}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{labs}\NormalTok{(}\AttributeTok{X =} \StringTok{"Sample\_ID"}\NormalTok{ , }\AttributeTok{y =} \StringTok{"Proportion"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{BCB-546-R_files/figure-latex/unnamed-chunk-8-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\AttributeTok{filename =} \StringTok{"./Graphs\_files/Zygozity in the sampleID.png"}\NormalTok{ , }\AttributeTok{device =} \StringTok{"png"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Saving 6.5 x 4.5 in image
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\# SNPs / Zygosity in different groups}
\FunctionTok{ggplot}\NormalTok{(tidy\_fang, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ Group , }\AttributeTok{fill =}\NormalTok{ new\_sequence)) }\SpecialCharTok{+} \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{position =} \StringTok{"fill"}\NormalTok{) }\SpecialCharTok{+} 
  \FunctionTok{theme\_bw}\NormalTok{() }\SpecialCharTok{+} \FunctionTok{theme}\NormalTok{(}\AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{5}\NormalTok{, }\AttributeTok{angle =} \DecValTok{90}\NormalTok{))}\SpecialCharTok{+} \FunctionTok{labs}\NormalTok{(}\AttributeTok{y =} \StringTok{"Proportion"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{BCB-546-R_files/figure-latex/unnamed-chunk-8-2.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\AttributeTok{filename =} \StringTok{"./Graphs\_files/Zygosity{-}SNP proportion in different groups.png"}\NormalTok{ , }\AttributeTok{device =} \StringTok{"png"}\NormalTok{ )}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Saving 6.5 x 4.5 in image
\end{verbatim}

\end{document}
